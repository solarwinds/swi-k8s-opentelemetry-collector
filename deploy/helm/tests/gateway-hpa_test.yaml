# yaml-language-server: $schema=https://raw.githubusercontent.com/helm-unittest/helm-unittest/main/schema/helm-testsuite.json
suite: Test for gateway-hpa
templates:
  - gateway/gateway-hpa.yaml
tests:
  - it: Gateway collector's HPA spec should match snapshot when using default values on cluster with Metrics Server
    template: gateway/gateway-hpa.yaml
    capabilities:
      apiVersions:
        - metrics.k8s.io/v1beta1
    asserts:
      - matchSnapshot:
          path: spec
  - it: Gateway collector's HPA should not be generated when using default values on cluster without Metrics Server
    template: gateway/gateway-hpa.yaml
    asserts:
      - hasDocuments:
          count: 0
  - it: Gateway collector's HPA should not be generated when the Gateway collector is disabled
    template: gateway/gateway-hpa.yaml
    capabilities:
      apiVersions:
        - metrics.k8s.io/v1beta1
    set:
      otel.gateway.enabled: false
    asserts:
      - hasDocuments:
          count: 0
