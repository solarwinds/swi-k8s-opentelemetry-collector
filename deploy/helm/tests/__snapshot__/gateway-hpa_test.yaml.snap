Gateway collector's HPA spec should match snapshot when using default values on cluster with Metrics Server:
  1: |
    maxReplicas: 1
    metrics:
      - resource:
          name: cpu
          target:
            averageUtilization: 80
            type: Utilization
        type: Resource
      - resource:
          name: memory
          target:
            averageUtilization: 80
            type: Utilization
        type: Resource
    minReplicas: 1
    scaleTargetRef:
      apiVersion: apps/v1
      kind: Deployment
      name: RELEASE-NAME-swo-k8s-collector-gateway
